partiql-eval/src/main/java/org/partiql/eval/value/Float32Value.java

```diff
@@ -0,0 +1,32 @@
+package org.partiql.eval.value;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.value.PartiQLValueType;
+
+/**
+ * This shall always be package-private (internal).
+ */
+class Float32Value implements PQLValue {
+
+    private final float _value;
+
+    Float32Value(float value) {
+        _value = value;
+    }
+
+    @Override
+    public boolean isNull() {
+        return false;
+    }
+
+    @Override
+    public float getFloat() {
+        return _value;
+    }
+
```

Comment:
This is starting to look exactly like PartiQLValue just in Java and with the `getFloat()` .. `getLong()` etc. I have a couple of questions.

- What's the difference now? (besides flat class hierarchy)
- Why don't we have the coercions in the PQLValue? (in this case `getDouble()`)

Perhaps its been too long since I looked, but I don't think I understand the value prop here.

Reply:
Yes, this PR aims to replace PartiQLValue incrementally. I'd suggest:
1. Add PQLValue without releasing.
2. Update functions to use PQLValue. (subsequent PR)
3. Move the existing PartiQLValue to the plan package (maybe called PlanLiteral) (since it's use is more for the DOM)
4. Rename PQLValue to PartiQLValue

The difference is that we are moving away from using Java's type semantics in favor of our own type system. With this, we can/should be using our own type information to determine how to extract the underlying data. The idea is that this logic should be exposed by `partiql-eval` instead of `partiql-types`.

There shouldn't be a need for coercions that are not represented in the plan. What I'd like to see is that when the executor receives a plan, without any modification, the executor shall execute the plan. If a JVM double is expected, then the planner should have inserted the appropriate coercion.

Consider `Rex.Op.Path.Index`. The planner currently allows all integer types as the index, and then the executor is actually under the hood coercing any integer value into a JVM int. In my opinion, the "function signature" of the path index should have required that the index be a PartiQL int_32. With this mindset, the expressions representing the index (say its typed to be an INT16) would be wrapped with a coercion by the planner. We're currently making up for our lack of planning by adding all of these additional coercions in the execution.

Reply:
Discussed other points offline to create more distance from DOM.

> There shouldn't be a need for coercions that are not represented in the plan. What I'd like to see is that when the executor receives a plan, without any modification, the executor shall execute the plan. If a JVM double is expected, then the planner should have inserted the appropriate coercion.

Agreed, makes sense now. Thank you.